### PR TITLE
feat: jumpstart model id suggestions

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*

*Description of changes:*
This PR improves the error message when a bad model id is inputted for JumpStart utilities. If the model id does not exist for any versions, the error message indicates a closest match model id which can be used. 

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/cache.py [hunk 1]
```diff
@@ -211,7 +212,23 @@ def _get_manifest_key_from_model_id_semantic_version(
                 f"{model_version_to_use_incompatible_with_sagemaker} of {model_id}."
             )
             raise KeyError(error_msg)
-        error_msg = f"Unable to find model manifest for {model_id} with version {version}."
+
+        error_msg = f"Unable to find model manifest for {model_id} with version {version}. "
+
+        other_model_id_version = self._select_version(
+            "*", versions_incompatible_with_sagemaker
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/cache.py:219]: can you please add a comment as to why you use this variable.

File: src/sagemaker/jumpstart/cache.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 """This module defines the JumpStartModelsCache class."""
 from __future__ import absolute_import
 import datetime
+from difflib import get_close_matches
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/cache.py:16]: nice find!

**Review by JGuinegagne — CHANGES_REQUESTED** [review]
Minor comments/doc changes.

Also a question on how many model IDs suggestion do we surface.

File: src/sagemaker/jumpstart/cache.py [hunk 3]
```diff
@@ -211,7 +212,23 @@ def _get_manifest_key_from_model_id_semantic_version(
...
+        )
+        if other_model_id_version is not None:
+            error_msg += (
+                f"Consider using model id {model_id} with version "
+                f"{other_model_id_version}."
+            )
+
+        else:
+            possible_model_ids = [header.model_id for header in manifest.values()]
+            closest_model_id = get_close_matches(model_id, possible_model_ids, n=1, cutoff=0)[0]
```
[the remark this answers is not in the document]
Comment (evakravi) [reply src/sagemaker/jumpstart/cache.py:229]: Let's just do 1 and make it simple. If we want to make this really good, we're better off just having a separate utility for searching model ids.

[the remark this answers is not in the document]
Comment (JGuinegagne) [reply src/sagemaker/jumpstart/cache.py:229]: another question, can you get an `IndexError` here if `get_close_matches` returns an empty list?

If that's possible, could you handle this edge case and add a unit test for it?

[the remark this answers is not in the document]
Comment (evakravi) [reply src/sagemaker/jumpstart/cache.py:229]: No, because the cutoff is 0, there will always be a match.

[the remark this answers is not in the document]
Comment (JGuinegagne) [reply src/sagemaker/jumpstart/cache.py:229]: Thanks. I know this is super edge case, but could there be a case where `possible_model_ids` is an empty list?

File: src/sagemaker/jumpstart/cache.py (hunk 1 above)
Comment (JGuinegagne) [reply src/sagemaker/jumpstart/cache.py:219]: reiterate: please explain why we use a variable called `versions_incompatitble_with_sagemaker` here.

File: src/sagemaker/jumpstart/cache.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (evakravi) [reply src/sagemaker/jumpstart/cache.py:233]: Only if the manifest is empty

**Review by JGuinegagne — APPROVED** [review]

**Review by shreyapandit — APPROVED** [review]

**Review by navinsoni — APPROVED** [review]
